src/llscan.cc

```diff
@@ -1141,6 +1164,51 @@ uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t word) {
 }
 
 
+void FindJSObjectsVisitor::InsertOnMapsToInstances(
+    uint64_t word, v8::Map map, FindJSObjectsVisitor::MapCacheEntry map_info,
+    v8::Error& err) {
+  TypeRecord* t;
+
+  auto entry = std::make_pair(map_info.type_name, nullptr);
+  auto pp = &llscan_->GetMapsToInstances().insert(entry).first->second;
+  // No entry in the map, create a new one.
+  if (*pp == nullptr) *pp = new TypeRecord(map_info.type_name);
+  t = *pp;
+
+  // Determine if this is a new instance.
+  // (We are scanning pointers to objects, we may have seen this location
+  // before.)
+  if (t->GetInstances().count(word) == 0) {
+    t->AddInstance(word, map.InstanceSize(err));
+  }
+}
+
+void FindJSObjectsVisitor::InsertOnDetailedMapsToInstances(
+    uint64_t word, v8::Map map, FindJSObjectsVisitor::MapCacheEntry map_info,
+    v8::Error& err) {
+  DetailedTypeRecord* t;
+
+  auto type_name_with_properties = map_info.GetTypeNameWithProperties();
+
+  auto entry = std::make_pair(*type_name_with_properties, nullptr);
+  auto pp = &llscan_->GetDetailedMapsToInstances().insert(entry).first->second;
+  // No entry in the map, create a new one.
+  if (*pp == nullptr)
+    *pp = new DetailedTypeRecord(*map_info.GetTypeNameWithProperties(
+                                     MapCacheEntry::kDontShowArrayLength, 3),
```

Comment:
It might worth making `3` a static variable in `FindJSObjectsVisitor` (ideally it can be an option)